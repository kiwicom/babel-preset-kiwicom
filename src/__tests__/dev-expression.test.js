// @flow

import * as babel from '@babel/core';
import { generateTestsFromFixtures } from '@kiwicom/test-utils';

function transform(input) {
  const oldEnv = process.env.NODE_ENV;
  process.env.NODE_ENV = '';

  const transformedCode = babel.transform(input, {
    plugins: [require('../dev-expression')],
  }).code;

  process.env.NODE_ENV = oldEnv;
  return transformedCode;
}

generateTestsFromFixtures(
  `${__dirname}/__fixtures-dev-expression__`,
  transform,
);
