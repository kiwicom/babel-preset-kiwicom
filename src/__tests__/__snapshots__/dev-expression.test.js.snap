// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: condition.js 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
// @flow

if (__DEV__) {
  console.log('foo'); // eslint-disable-line no-console
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
"use strict";

if (process.env.NODE_ENV !== "production") {
  console.log('foo'); // eslint-disable-line no-console
}
`;

exports[`matches expected output: objectKey.js 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
// @flow

declare var __DEV__: string;

module.exports = {
  rules: {},
  globals: {
    __DEV__: 'value',
    [__DEV__]: "this can be changed because it's dynamic",
    key: __DEV__,
  },
};

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
"use strict";

module.exports = {
  rules: {},
  globals: {
    __DEV__: 'value',
    [process.env.NODE_ENV !== "production"]: "this can be changed because it's dynamic",
    key: process.env.NODE_ENV !== "production"
  }
};
`;

exports[`matches expected output: ternary.js 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
// @flow

// eslint-disable-next-line no-unused-vars
const dev = __DEV__ ? 'true' : 'false';

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
"use strict";

// eslint-disable-next-line no-unused-vars
const dev = process.env.NODE_ENV !== "production" ? 'true' : 'false';
`;
